### PR TITLE
Adding tests and changing controllers to  add busines logic of only g…

### PR DESCRIPTION
**Pull request by amayer1**
…etting repos you own or public repos

### CODE REVIEW

File: boosters/node-beginner/src/api/controllers/repoController.js [hunk 1]
```diff
@@ -26,9 +32,10 @@ const createRepo = async (req, res) => {
 
 const getRepo = async (req, res) => {
   const { repoId } = req.params;
-
+  const { userId } = req.body;
   const repo = await Repos.findByPk(repoId);
-  res.send(repo);
+
+  repo.userId === userId || repo.isPrivate === false ? res.send(repo) : res.send(null);
```
Comment (chao-henry) [inline boosters/node-beginner/src/api/controllers/repoController.js]: Should it return a 401 if the request is not authorized?

File: boosters/node-beginner/tests/07_exercise.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+describe("Not sure yet", () => {
+  beforeAll(async () => {
+    await setup("Users", users)();
+    return setup("Repos", repos)();
+  });
+
+  afterAll(teardown);
+
+  xit("Should return all public repositories and all repositories of current user", async () => {
```
Comment (chao-henry) [inline boosters/node-beginner/tests/07_exercise.test.js:32]: Pick one user to be the current user, then validate that all public and owned repositories are returned.

File: boosters/node-beginner/tests/07_exercise.test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        curr1.body.reduce(
+          (acc, curr) => acc && (users[index].id === curr.userId || !curr.isPrivate),
+          true
+        ),
+      true
+    );
+    expect(test).toEqual(true);
+  });
+
+  it("Should return a single repository if it's either public or owned by current user. Otherwise return null", async () => {
```
Comment (chao-henry) [inline boosters/node-beginner/tests/07_exercise.test.js]: Split this out into three test cases. One if it's public, another if it's owned by the current user, and one where the repo is not returned.

**Review by chao-henry — CHANGES_REQUESTED** [review]

File: boosters/node-beginner/src/api/controllers/repoController.js (hunk 1 above)
Comment (amayer1) [reply boosters/node-beginner/src/api/controllers/repoController.js]: Probably - I was just following the graphQL one where null was  returned, but I think setting status code is better

**Review by chao-henry — APPROVED** [review]

**Pull request merged**